### DISCUSSION TITLE
Is it possible to not use unleash UI?

### DISCUSSION THREAD

**Discussion by makeavish** [Unleash/unleash#2080] [Q&A] [answered]
We are a self hosted opensource APM tool and want to integrate feature toggling through our own UI, wanted to know if it's possible to use Unleash SDK and API but not using Unleash UI?

**Comment by thomasheartman** [comment, marked as answer]
Yes, that is perfectly possible! 😄 Everything in the Unleash UI goes via the Unleash API, so whatever the UI can do, you can also do directly with the API. 

If you're looking for an overview over all the endpoints, I'd suggest checking out your instance's OpenAPI spec (refer to our [how-to enable the OpenAPI spec guide](https://docs.getunleash.io/how-to/how-to-enable-openapi) for the steps). We are also working on an [OpenAPI spec integration with the docs](https://docs.getunleash.io/reference/api/unleash), which might be of use. It's not perfectly polished yet, but we're working on it. Note, however, that it lists _all_ Unleash endpoints including ones that are restricted to Pro and Enterprise version, and that it currently only shows the latest version. So it's a good place to go for reference, but your own instance's spec will always be more accurate for your use case.
